### PR TITLE
feat: Calendar supports `showWeek` attribute

### LINKED ISSUE #49062: Add boolean prop ```showWeekNumber``` to ```Calendar``` component to display the week number on the fullscreen calendar
### What problem does this feature solve?

It would be very useful if the fullscreen calendar could display the week number in the first column, instead of having to add custom logic to display it every time I use this component.

### What does the proposed API look like?

| Property | Description | Type | Default |
| --- | --- | --- | --- |
| showWeekNumber | Display the week number on the first column of the calendar when fullscreen is `true` | boolean | false |

### PR DESCRIPTION
**Pull request by nikzanda**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [x] 🆕 New feature
- [ ] 🐞 Bug fix
- [x] 📝 Site / documentation improvement
- [x] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

Close #49062

### 💡 Background and Solution

By adding the boolean property `showWeekNumber`, the fullscreen calendar can display the week number in the first cell of each week.

### 📝 Change Log

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Calendar supports `showWeek` attribute |
| 🇨🇳 Chinese | Calendar 支持 `showWeek` 属性 |

### CODE REVIEW

File: components/calendar/demo/week-number.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## zh-CN
+
+通过将 `showWeekNumber` 属性设置为 `true`，在全屏日历中显示周数。
```
Comment (afc163) [inline components/calendar/demo/week-number.md:3]: showWeekNumber => showWeek

**Comment by afc163** [comment]
showWeekNumber => showWeek

File: components/calendar/demo/week-number.md (hunk 1 above)
Comment (afc163) [reply components/calendar/demo/week-number.md:3]: Yes, DatePicker use `showWeek` as prop name: https://ant.design/components/date-picker

Comment (nikzanda) [reply components/calendar/demo/week-number.md]: Ok, I have renamed the prop as requested. Can you check now if the PR can be merged? Thanks.

File: components/calendar/index.zh-CN.md [hunk 2]
```diff
@@ -46,6 +47,7 @@ coverDark: https://mdn.alipayobjects.com/huamei_7uahnr/afts/img/A*-p-wQLik200AAA
 | defaultValue | 默认展示的日期 | [dayjs](https://day.js.org/) | - |  |
 | disabledDate | 不可选择的日期，参数为当前 `value`，注意使用时[不要直接修改](https://github.com/ant-design/ant-design/issues/30987) | (currentDate: Dayjs) => boolean | - |  |
 | fullscreen | 是否全屏显示 | boolean | true |  |
+| showWeek | 是否在第一个星期几单元格上显示周数 | boolean | false |  |
```
Comment (thinkasany) [inline components/calendar/index.zh-CN.md:50]:
```suggestion
| showWeek | Calendar 下展示当前周 | boolean | false | 5.23.0 |
```

File: components/calendar/index.en-US.md [hunk 3]
```diff
@@ -45,6 +46,7 @@ Common props ref：[Common props](/docs/react/common-props)
 | defaultValue | The date selected by default | [dayjs](https://day.js.org/) | - |  |
 | disabledDate | Function that specifies the dates that cannot be selected, `currentDate` is same dayjs object as `value` prop which you shouldn't mutate it](https://github.com/ant-design/ant-design/issues/30987) | (currentDate: Dayjs) => boolean | - |  |
 | fullscreen | Whether to display in full-screen | boolean | true |  |
+| showWeek | Whether to display week number on first week day cell | boolean | false |  |
```
Comment (thinkasany) [inline components/calendar/index.en-US.md:49]:
```suggestion
| showWeek | Show week info when in Calendar| boolean | false | 5.23.0 |
```

File: components/calendar/index.en-US.md [hunk 4]
```diff
@@ -18,6 +18,7 @@ When data is in the form of dates, such as schedules, timetables, prices calenda
 <code src="./demo/notice-calendar.tsx" clientOnly>Notice Calendar</code>
 <code src="./demo/card.tsx" clientOnly>Card</code>
 <code src="./demo/select.tsx" clientOnly>Selectable Calendar</code>
+<code src="./demo/week-number.tsx" clientOnly>Week number</code>
```
Comment (thinkasany) [inline components/calendar/index.en-US.md:21]:
```suggestion
<code src="./demo/week-number.tsx" clientOnly version="5.23.0">Week number</code>
```

File: components/calendar/index.zh-CN.md [hunk 5]
```diff
@@ -19,6 +19,7 @@ coverDark: https://mdn.alipayobjects.com/huamei_7uahnr/afts/img/A*-p-wQLik200AAA
 <code src="./demo/notice-calendar.tsx" clientOnly>通知事项日历</code>
 <code src="./demo/card.tsx" clientOnly>卡片模式</code>
 <code src="./demo/select.tsx" clientOnly>选择功能</code>
+<code src="./demo/week-number.tsx" clientOnly>周数</code>
```
Comment (thinkasany) [inline components/calendar/index.zh-CN.md:22]:
```suggestion
<code src="./demo/week-number.tsx" clientOnly version="5.23.0">周数</code>
```

File: components/calendar/index.en-US.md (hunk 4 above)
Comment (thinkasany) [reply components/calendar/index.en-US.md:21]: The demo that adds the version will have ribbons showing the version.
image

File: components/calendar/generateCalendar.tsx [hunk 6]
```diff
@@ -182,22 +186,45 @@ const generateCalendar = <DateType extends AnyObject>(generateConfig: GenerateCo
...
+            generateConfig.getWeekDay(date)
+        ) {
+          const checkIsSameMonth = isSameMonth(date, mergedValue || today, generateConfig);
+          const weekNumber = generateConfig.locale.getWeek(info.locale!.locale, date);
+          weekNumberNode = checkIsSameMonth ? (
+            <Typography.Text type="secondary">{weekNumber}</Typography.Text>
+          ) : (
+            weekNumber
+          );
+        }
```
Comment (afc163) [inline components/calendar/generateCalendar.tsx:203]: You don't need to create week cell here, passing `showWeek` prop to RcPicker would be working.

图片

Then styling these cells.

图片

**Comment by thinkasany** [comment]
I think the pr title is too long and can be shortened.  I suggest changing to ```feat: Calendar supports `showWeek` attribute```

And the changelog should also be modified, which can help reviewers organize the changelog.

en:  ```feat: Calendar supports `showWeek```
zh:  ```feat: Calendar 支持 `showWeek` 属性```

image

image

image

File: components/calendar/generateCalendar.tsx (hunk 6 above)
Comment (nikzanda) [reply components/calendar/generateCalendar.tsx:203]: @thinkasany Thank you for the help!

@afc163 By implementing the code from thinkasany, the week number appears in the same style as the day number, as shown in the screenshot. However, this approach adds a new cell that occupies unnecessary space and is also selectable, which I don't find necessary for the fullscreen calendar.
Given this, can you confirm that the approach you suggest is the best way to implement this feature?

Additionally, I am not familiar with the Ant Design code. Could you assist me with the style implementation? Thank you for your help.

Comment (nikzanda) [reply components/calendar/generateCalendar.tsx:203]: @afc163 Hi, any news on this PR? Can you help me to complete the task? Thanks.

**Comment by yoyo837** [comment]
New features should be send to the `feature` branch.

File: components/calendar/generateCalendar.tsx (hunk 6 above)
Comment (nikzanda) [reply components/calendar/generateCalendar.tsx:203]: Any news?

Comment (afc163) [reply components/calendar/generateCalendar.tsx:203]: @nikzanda Here is the implementation based on my suggestion above: https://github.com/ant-design/ant-design/pull/52072/

Could you review it?

Comment (afc163) [reply components/calendar/generateCalendar.tsx:203]: I prefer to keep same behavior with DatePicker: https://ant.design/components/date-picker#date-picker-demo-basic

图片

**Pull request closed without merging**